### PR TITLE
Choose postage on POST request for precompiled letters

### PR DESCRIPTION
**Pull request by CrystalPea**
This PR enables adding optional 'postage' attribute to a POST request to 'post_precompiled_letter_notification' endpoint and sets notification postage based on this attribute if it's present. The attribute value needs to be either 'first' or 'second'.

Pivotal ticket: https://www.pivotaltracker.com/story/show/160231957

### CODE REVIEW

File: app/notifications/process_notifications.py [hunk 1]
```diff
@@ -112,11 +113,14 @@ def persist_notification(
...
-        template = dao_get_template_by_id(template_id, template_version)
-        if service.has_permission(CHOOSE_POSTAGE) and template.postage:
-            notification.postage = template.postage
+        if postage:
+            notification.postage = postage
         else:
-            notification.postage = service.postage
+            template = dao_get_template_by_id(template_id, template_version)
+            if service.has_permission(CHOOSE_POSTAGE) and template.postage:
+                notification.postage = template.postage
```
Comment (servingUpAces) [inline app/notifications/process_notifications.py]: The template.postage should be passed into this method, we already have the template before we call this method.  That will save us a db call.

Comment (CrystalPea) [reply app/notifications/process_notifications.py]: I can do it, but then I would have to add that argument in many places that call `persist_notification`, and check that all those places have template object:

screen shot 2019-01-11 at 14 40 18

I didn't introduce this call to db for this particular story, I just changed indentation.

Comment (CrystalPea) [reply app/notifications/process_notifications.py]: Anyway, let me know.

Comment (servingUpAces) [reply app/notifications/process_notifications.py]: If we can avoid a call to the db we should.

Comment (CrystalPea) [reply app/notifications/process_notifications.py]: ok, this has been done now 👍

**Pull request merged**